Feature/examples

luispresuelVenafi opened the pull request.

Commit:
F5 BIG-IP example first README draft

Commit:
removes tfvars file

Commit:
Fixes block codes format

marcos-albornoz commented on examples/README.md:
```diff
@@ -0,0 +1,50 @@
+# Venafi Provider Examples
+
+In this section you'll find examples to make your implementation of your venafi provider for automating your process of SSL integration in your infrastructure.
```
Venafi Provider for Terraform

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+# Build a set of servers with a balancer using F5
+
+This example will guide you in mounting a load balancer server with 3 pool members of an f5 BIG-IP instance and make certificates for those sites using venafi's product terraform implentation in oder to provide [SSL termination](https://www.techwalla.com/articles/what-is-ssl-termination).
```
*implementation

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+
+## Personas
+
+The steps described in this example are typically performed by a **System Administrator** or a **DevOps engineer**.
```
*DevOps Engineer or Systems Administrator

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+
+## Scenario
+
+In order to increase reliability and capacity of applications, a load balancer manages web traffic of your server application into child nodes in order to reduce the "weight load" of those applications.
```
The "child" word is unneeded

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+
+We will divide the process in the following steps:
+
+> **_Note:_**  This steps are providing a suggested terraform file structure for this example only but you could still use the same configuration as your liking.
```
change "liking" for "prefer"

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+
+> **_Note:_**  This steps are providing a suggested terraform file structure for this example only but you could still use the same configuration as your liking.
+
+1. Create your terraform variables file
```
Put dots at the end.

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+
+## Scenario Introduction
+
+As for this example scenario, you'll generate a certificate for ``demo-f5-bigip.venafi.example`` using Venafi Provider for Hashicorp terraform using Venafi Cloud - DevOps Accelerate. Thus adding them to your F5 BIG-IP resources, then use them in the load balancer Node, and, finally, you'll configure the "pool" for you load balancer nodes.
```
Remove the "DevOps Accelerate" mention

marcos-albornoz commented on examples/f5_example/README.md:
```diff
@@ -0,0 +1,274 @@
+- The configuration for your site
+- The F5 partition where your data is stored
+- The Virtual IP and Port which is the entry point for your traffic-management object of your virtual server
+- The pool memebers are physical nodes on the network (NGINX servers for this example).
```
*members

marcos-albornoz commented on examples/f5_example/README.md:
````diff
@@ -0,0 +1,274 @@
+    }
+    ```
+
+2. Set your asset_name for your vars in `locals`, remeber that locals<sup>[1](https://www.terraform.io/docs/language/values/locals.html)</sup> are values that multiple times within a module without repeating it:
````
*remember

marcos-albornoz commented on examples/f5_example/README.md:
````diff
@@ -0,0 +1,274 @@
+    }
+    ```
+
+5. Create your pool members resources to manage memebership in pools<sup>[3](https://registry.terraform.io/providers/F5Networks/bigip/latest/docs/resources/bigip_ltm_pool_attachment)</sup>:
````
membership

Commit:
Fixes typos in README. Modifies structure and adds some links accordingly. Adds tpp use case

Commit:
Fixes some other typoes and some formatting in the examples' README file

Commit:
typo: this -> these

luispresuelVenafi committed:
Fixes type and chages order of the way of displaying the venafi cloud and tpp variable settings

luispresuelVenafi committed:
Some other fixes to the README. Modifies provided example files accordingly

cdmadrigal committed:
Initial Sample with AWS ELB

Initial sample. Creates a VPC, EC2 instance and ALB bounded to a certificate issued via Venafi. Initial sample was tested on V0.10.2 of the Venafi Terraform provider. Tested on Venafi TPP 20.1.7.

Note: Certain IAM permission may be necessary to create all the resources in the example. (VPC, Networking, Compute, Certificates)

asquith-venafi committed:
fix typo: partion -> partition

luispresuelVenafi committed:
Fixes some typos. Removes hyperlink from sceneario image. Adds ASCIICINEMA videos. Changes load balancer term to ADC. Adds capital letters wher needed

luispresuelVenafi committed:
Merge branch 'feature/examples' of https://github.com/Venafi/terraform-provider-venafi into feature/examples

luispresuelVenafi committed:
fixes url and rebase from commit

luispresuelVenafi committed:
Merge pull request #38 from cdmadrigal/feature/examples

Initial Sample with AWS ELB

luispresuelVenafi committed:
Renames folder to fit folder format

luispresuelVenafi committed:
Citrix ADC doc improvements

luispresuelVenafi committed:
Adds example files for citrix adc. Changes base location outside of citrix example.

jdw2465VEN committed:
Minor edits by UA to test process for reviews

jdw2465VEN committed:
Merge pull request #46 from jdw2465VEN/patch-1

Minor edit by UA to test process for reviews

luispresuelVenafi committed:
deletes some unneeded files. Elaborates more about Terraform's filesystem mirror. Adds some changes to the Citrix provided default files. Adds recordings for Citrix docs

luispresuelVenafi committed:
Merge branch 'feature/examples' of https://github.com/Venafi/terraform-provider-venafi into feature/examples

luispresuelVenafi committed:
Fixes some typos. Changes titles. Fixes some formmating in provided files for Citrix docs

luispresuelVenafi committed:
changes terraform.vars to example format for AWS example. Adds new lines to to files that were missing

luispresuelVenafi committed:
Updates recordings since previous ones were archived

jdw2465VEN committed:
Update README.md

Adding the first of what will be my general feedback for @luispresuelVenafi about ways to make the document easier for customers to follow

jdw2465VEN committed:
Update README.md

additional clarification added to my first comment

jdw2465VEN committed:
Update README.md

removed my comment as I'm going to create a new branch and make my comments there...

jdw2465VEN committed:
Update README.md

First set of comments and changes; checking in now because I'll be working through this

jdw2465VEN committed:
Merge pull request #47 from Venafi/UA-Feedback-patch-1

Update README.md

jdw2465VEN committed:
Update README.md

jdw2465VEN committed:
Update README.md

A few more additions

jdw2465VEN committed:
Update README.md

jdw2465VEN committed:
Update README.md

jdw2465VEN committed:
Update README.md

jdw2465VEN committed:
Update README.md

More comments and updates...

jdw2465VEN committed:
Update README.md

Comments and changes...more to come...

jdw2465VEN committed:
Update README.md

More edits...this time to the prereq content

jdw2465VEN committed:
Update README.md

More comments/suggestions

jdw2465VEN committed:
Update README.md

More comments/suggestions

jdw2465VEN committed:
Update README.md

Final comments; next step is to make this into a final draft to prep for publishing it

jdw2465VEN committed:
Update README.md

Moved section content around to match our structure

jdw2465VEN committed:
Update README.md

Last commit I'll make until copy edit time

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+<!-- 
+ORIGINAL TEXT: This example will guide you in mounting a [CITRIX-ADC](https://www.citrix.com/products/citrix-adc/) instance and make certificates for those sites using Venafi's product [HashiCorp Terraform](https://terraform.io/) implementation in order to provide [SSL termination](https://www.techwalla.com/articles/what-is-ssl-termination). -->
+<!--
+DW: Hi Luis, I think this first para should describe the desired outcome: state in simple terms what the user can expect when they implement your example. In fact, I re-wrote your title to state more directly what the desired outcome actually is (as I understand it): securing application delivery using Venafi Provider for HashiCorp Terraform with my Citrix ADC instance. So for example, while moun ...
```
Sure, it makes sense. Looks good to me

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+DW: Hi Luis, I think this first para should describe the desired outcome: state in simple terms what the user can expect when they implement your example. In fact, I re-wrote your title to state more directly what the desired outcome actually is (as I understand it): securing application delivery using Venafi Provider for HashiCorp Terraform with my Citrix ADC instance. So for example, while moun ...
+-->
+
+## Who should use this example? <!-- Suggest not using "Persona" as this is an UX term used more internally in software dev than a term that users would typically understand; while more techy people will use your example and might understand UX notion of personas, they are in this context a user. -->
```
LGTM

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+The steps described in this example are typically performed by **DevOps engineers** or **system administrators**. Generally, you'll need a basic understanding of Citrix ADC, Venafi Trust Protection Platform or Venafi Cloud, and the required permissions for completing the tasks described in the example.
+
+<!--
+DW: So I suggest adding--as I tried to do in that second sentence--the basic knowledge (as well as the permissions and access to the various systems) that is required in order to successfully complete your example. 
```
LGTM, Just a comment. In this Citrix scenario you should also have some basic knowledge of Bash command language in order to set your provider locally. Should this also be included here? Or should we give it for granted? 

This is something to consider that may happen in the future since not all Terraform providers may not be registered in Terraform Registry at that time (hence that's why we provided a "base installation for providers to be installed locally").

jdw2465VEN replied:
Good point. I suggest you include it as a simple TIP-style note...something like: 

>**TIP** Having at least some basic knowledge of the Bash command language is helpful, such as when you need to set your provider locally.

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+<!--
+DW: So I suggest adding--as I tried to do in that second sentence--the basic knowledge (as well as the permissions and access to the various systems) that is required in order to successfully complete your example. 
+-->
+## About this example <!--To make this more conversational and friendly, I've changed the title from "Solution" to this one. -->
```
LGTM

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+
+<!-- ORIGINAL TEXT: In order to increase reliability and capacity of applications, an application delivery controller(ADC) manages web traffic of your server application into nodes in order to reduce the "weight load" of those applications.
+-->
+<!-- DW: I took this first para out because I don't think we need to describe in this section what ADCs are and what they do. -->
```
LGTM

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+
+In this example, we use Terraform's _infrastructure as code_ automation process with the _Venafi Provider_ to generate and install certificates as part of SSL termination on a load balancer (Citrix ADC). We'll also utilize three HTTP servers contained in a cluster as the endpoints that are sending and receiving web traffic and being managed by Citrix ADC.
+
+<!-- **DW:** The original paragraph above wasn't clear to me; in my attempt to undersand it, I've written a new para. If I've lost the technical meaning, it's because I couldn't follow the original logic. Some of the questions I had from the original were these: Which parts of the explanation are Terraform's and which parts are Venafi...because the first half of the original sentence made it soun ...
```
We should change the "load balancer" to "application delevery controller"  or "ADC" since this is modern term for load balancer. Perhaps we could rephrase it from:

"to generate and install certificates as part of SSL termination on a load balancer (Citrix ADC)"

to:

"to generate and install certificates as part of SSL termination on a Application Delevery Controller (Citrix ADC) for load balancing web traffic."

jdw2465VEN replied:
I agree, Luis...with just a couple of minor edits:

> . . . to generate and install certificates as part of SSL termination on an ADC (specifically, Citrix ADC) for load balancing web traffic. 

Because we already defined application delivery controller as ADC earlier in the document, I used ADC here, per our style guide.

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+
+<!-- **DW:** The original paragraph above wasn't clear to me; in my attempt to undersand it, I've written a new para. If I've lost the technical meaning, it's because I couldn't follow the original logic. Some of the questions I had from the original were these: Which parts of the explanation are Terraform's and which parts are Venafi...because the first half of the original sentence made it soun ...
+
+<!-- NOTE: DW moved the content below from what was called the Scenario Introduction section, which was below the Getting Started section, per our discussion on 31 March; we'd agreed that it makes more sense here and that it interupted the flow of the document in it's old location. --> 
```
LGTM

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+### About retrieving a certificate using the _Venafi Provider for Terraform_
+
+> **NOTE** The only purpose of the credentials used in this example is illustrative, in a real life scenario they must be considered as **weak** and **insecure**.
+<!-- This seems like a strange place for this note; is this about the generic creds used in the steps below? And is the intent to tell users that they shouldn't use simple passwords (e.g. "password") in production environments? Once I understand the purpose, I can suggest some changes... -->
```
This has to do with the development environment for this example. In a production enviroment, the credentials would signed by a trusted CA, but in this case the credentials are self-signed by the platform you are requesting the credentials:

![venafi-cloud-example](https://user-images.githubusercontent.com/77068025/113330899-7cd92c80-92dc-11eb-8db8-e13b28bce0a2.png)

And self-signed credentials are considered dangerous and insecure:

https://www.globalsign.com/en/ssl-information-center/dangers-self-signed-certificates

jdw2465VEN replied:
Great point. So borrowing from our own documentation on self-signed certs, I suggest you add a note like this:

>**Best Practice** In general, be careful when using self-signed certificates because of the inherent risks of no identity verification or trust control. The public and private keys are both held by the same entity. Also, self-signed certificates cannot be revoked; they can only be replaced. If an attacker has already gained access to a system, the attacker can spoof the identity of the subject. Of course, CAs can revoke a certificate only when they discover the compromise.

luispresuelVenafi commented on examples/citrix_adc/README.md:
```diff
@@ -0,0 +1,369 @@
+[![asciicast](https://asciinema.org/a/PrCtLI7cwkZC6RUriqpwuiQVU.svg)](https://asciinema.org/a/PrCtLI7cwkZC6RUriqpwuiQVU)
+
+## What's next
+<!-- should keep this section brief; if the answer is more than a small paragraph, I suggest that you link to another article/topic/website somewhere -->
```
Got it

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+ORIGINAL TEXT: This example will guide you in mounting a [F5 BIG-IP](https://www.f5.com/products/big-ip-services) instance and make certificates for those sites using Venafi's product [HashiCorp Terraform](https://terraform.io/) implementation in order to provide [SSL termination](https://www.techwalla.com/articles/what-is-ssl-termination). 
+-->
+<!--
+DW: Hi Ricardo, I think this first para should describe the desired outcome: state in simple terms what the user can expect when they implement your example. In fact, I re-wrote your title to state more directly what the desired outcome actually is (as I understand it): securing application delivery using Venafi Provider for HashiCorp Terraform with my F5 BIG-IP instance. My thinking is that whil ...
```
LGTM.

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+
+The steps described in this example are typically performed by a **DevOps engineers** or **system administrators**. Generally, you'll need a basic understanding of F5 BIG-IP, Venafi Trust Protection Platform or Venafi Cloud, and the required permissions for completing the tasks described in the example.
+<!--
+DW: So I suggest adding--as I tried to do in that second sentence--the basic knowledge (as well as the permissions and access to the various systems) that is required in order to successfully complete your example. 
```
LGTM

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+
+In this example, we use Terraform's _infrastructure as code_ automation process with the _Venafi Provider_ to generate and install certificates as part of SSL termination on a load balancer (F5 BIG-IP). We'll also utilize three HTTP servers contained in a cluster as the endpoints that are sending and receiving web traffic that's being managed by F5 BIG-IP.
+<!-- 
+**DW:** The original paragraph above wasn't clear to me; in my attempt to undersand it, I've written a new para. If I've lost the technical meaning, it's because I couldn't follow the original logic. Some of the questions I had from the original were these: Which parts of the explanation are Terraform's and which parts are Venafi...because the first half of the original sentence made it sound lik ...
```
Same as: https://github.com/Venafi/terraform-provider-venafi/pull/35/files#r605820334

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+
+## About retrieving a certificate using the _Venafi Provider for Terraform_
+
+> **NOTE** The only purpose of the credentials used in this example is illustrative, in a real life scenario they must be considered as **weak** and **insecure**. <!-- This seems like a strange place for this note; is this about the generic creds used in the steps below? And is the intent to tell users that they shouldn't use simple passwords (e.g. "password") in production environments? Once I u ...
```
Same as: https://github.com/Venafi/terraform-provider-venafi/pull/35/files#r605824249

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+- Verify that you have administrator access to your F5 BIG-IP instance
+- Verify that you have three (3) web servers that are running your application; for this example, we'll use NGINX servers.
+
+## Getting started <!-- To give your document more of a flow forward, I changed the title from "Solution" to this one. Users love this title because it's like a sign-post letting them know that now we're getting down to business! -->
```
LGTM

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+<!-- Depending on your MD language, you could format these as expandable text so users can click the bullet item to reveal your answers. -->
+
+## Helpful references 
+<!-- I think if you use this idea of providing links to third-party resources, that we should make it part of our template; otherwise, I would move these links one at a time up into your content where it makes the most sense and is in context -->
```
Perhaps then we would go with "move these links to content" and remove this section, since it may not be able to be put in all examples as part of the template, because not all config options may have a direct link for the configuration (this if the provider is not registered in Terraform Registry).

jdw2465VEN replied:
I agree. So I suggest you move them to where they make sense; might be easier to see and manage them if you put them directly below the paragraph they relate to and format them in their own paragraph with something like this: 

>For more information, see [Using jet fuel to speed up your BMW](http://www.bmw.com).

luispresuelVenafi commented on examples/f5_bigip/README.md:
```diff
@@ -0,0 +1,353 @@
+[![asciicast](https://asciinema.org/a/V7TthUjHaejww5miuUbAnmyQS.svg)](https://asciinema.org/a/V7TthUjHaejww5miuUbAnmyQS)
+
+## What's next
+<!-- should keep this section brief; if the answer is more than a small paragraph, I suggest that you link to another article/topic/website somewhere -->
```
Got it.

luispresuelVenafi committed:
Attends more comments to the documentation review and updates accordingly. Removes unused code in Citrix example

jdw2465VEN committed:
Updated intro para 

This is to better match what we updated in the examples themselves

jdw2465VEN committed:
Copy edits complete

Also removed my old hidden comments

jdw2465VEN committed:
removed old hidden comments

jdw2465VEN committed:
and a few more

jdw2465VEN committed:
Copy edits and finishing touches

jdw2465VEN committed:
very minor tweaks here...take it or leave it

luispresuelVenafi committed:
Merge pull request #49 from jdw2465VEN/feature/examples

Feature/examples

luispresuelVenafi committed:
Fixe typo: aboout -> about
